src/client/js/settings.js

```diff
@@ -0,0 +1,156 @@
+const settingsAlertOptionsInputs = document.getElementsByClassName('js-settings-alert-options-input')
+
+if (settingsAlertOptionsInputs?.length) {
+  for (const inputElement of settingsAlertOptionsInputs) {
+    inputElement.addEventListener('change', async event => {
+      try {
+        const communicationOption = event.target.getAttribute('data-alert-option')
+        const csrfToken = document
+          .getElementById('js-settings')
+          .getAttribute('data-csrf-token')
+
+        const response = await fetch('/api/v1/user/update-comm-option', {
+          headers: {
+            'Content-Type': 'application/json',
+            'x-csrf-token': csrfToken
+          },
+          mode: 'same-origin',
+          method: 'POST',
+          body: JSON.stringify({ communicationOption })
+        })
+
+        if (response && response.redirected === true) {
+          throw response.error
+        }
+      } catch (err) {
+        throw new Error(`Updating communication option failed: ${err}`)
+      }
+      event.preventDefault()
+      return false
+    })
+  }
+}
+
+const settingsAddEmail = document.getElementById('js-settings-add-email-opener')
+if (settingsAddEmail) {
+  settingsAddEmail.addEventListener('click', async _ => {
+    try {
+      const addEmailDialog = document.getElementById('js-settings-modal')
+      addEmailDialog.showModal()
+    } catch (err) {
+      console.error(`Error: ${err}`)
+    }
+  })
+}
+
+const settingsClose = document.getElementById('js-settings-close')
+if (settingsClose) {
+  settingsClose.addEventListener('click', async _ => {
+    try {
+      const addEmailDialog = document.getElementById('js-settings-modal')
+      addEmailDialog.close()
+    } catch (err) {
+      console.error(`Error: ${err}`)
+    }
+  })
+}
+
+const settingsAddVerification = document.getElementById('js-settings-modal-send-verification')
+if (settingsAddVerification) {
+  settingsAddVerification.addEventListener('click', async _ => {
+    const email = document.getElementById('js-settings-email-modal-input').value
+    const csrfToken = document
+      .getElementById('js-settings')
+      .getAttribute('data-csrf-token')
+
+    const response = await fetch('/api/v1/user/email', {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-csrf-token': csrfToken
+      },
+      mode: 'same-origin',
+      method: 'POST',
+      body: JSON.stringify({ email })
+    })
+
+    if (!response?.ok) {
+      throw new Error(`Sending verification email failed: ${response?.error}`)
+    }
+
+    const addEmailDialogContent = document.getElementById('js-settings-modal-content')
+    // TODO: Localize string below
+    addEmailDialogContent.textContent = `Verify the link sent to ${email} to add it to Firefox Monitor. Manage all email addresses in Settings.`
```

Comment:
How should we best handle localized strings that are being set via JS — should we include `fluent`?

Reply:
I think so, @toufali might have a better idea.

Reply:
The `getMessage()` function in utils/fluent.js accepts an optional argument that should help cases like this.  Example:

https://github.com/mozilla/blurts-server/blob/fe9e0c31aa18bb0968311abaa4791136fbaeb1eb/src/views/partials/breaches.js#L75

https://github.com/mozilla/blurts-server/blob/fe9e0c31aa18bb0968311abaa4791136fbaeb1eb/locales/en/breaches.ftl#L10-L16


Here's another (possibly less complex) example where the string is translated into a constant rather than template:
https://github.com/mozilla/blurts-server/blob/fe9e0c31aa18bb0968311abaa4791136fbaeb1eb/src/views/partials/breaches.js#L37-L42

https://github.com/mozilla/blurts-server/blob/fe9e0c31aa18bb0968311abaa4791136fbaeb1eb/locales/en/breaches.ftl#L31-L33

Reply:
@toufali Thanks for the examples. My question here specifically was more related to: If we set localized strings from `JS` we’d need to include `fluent.js` in the frontend resources or is there another way we could do this?

Reply:
Ah I see.  

I'm not familiar with using fluent.js on the client – wouldn't that require special endpoints or localization server, and might that introduce lag in showing a proper translation?  Would be interested to hear more of this approach.

The legacy site appears to creates the modal with translations on the server, and then pass additional translation strings as needed via XHR response after user interaction:

https://github.com/mozilla/blurts-server/blob/1d59e3383aed8d672dd2b130f1e30b940536f12f/public/js/resolve-breaches.js#L116-L157

Other options:
- use a data-attribute on the relevant tag (if it exists) on initial render
- include the translations in an SSR `<script>` tag (though I believe this pollutes global scope, so likely not a first choice without additional thought) :
```js
<script>
var l10n = {
         addEmailDialogContent: 'a translated string'
      }
</script>
```

- you could use an SSR `<template>` tag to store strings for use on client:
```html
<template>
  <p data-dialog-txt='translated string'></p>
  <!-- or -->
  <p data-dialog-txt>translated string</p>
</template>
```

- expanding on the previous, you could just use a template element for the entire structure you intend to render dynamically, not just translations...  This might play nicely with the lightbox/dialog web component forthcoming this sprint – i.e. the client web component is a shell/wrapper that loads the SSR template content 🤷